platform/linux-generic/odp_timer.c

```diff
@@ -2140,5 +2158,18 @@ int _odp_timer_init_local(void)
 
 int _odp_timer_term_local(void)
 {
+#if CONFIG_TIMER_PROFILE_INLINE
+	if (timer_local.prof_rounds) {
+		int thr = odp_thread_id();
+
+		odp_ticketlock_lock(&timer_global->lock);
+		_ODP_PRINT("Inline timer profiling for thread %i:\n", thr);
+		_ODP_PRINT("  scan rounds:   %" PRIu64 "\n", timer_local.prof_rounds);
+		_ODP_PRINT("  ave scan nsec: %.1f\n",
+			   (double)timer_local.prof_nsec / timer_local.prof_rounds);
+		odp_ticketlock_unlock(&timer_global->lock);
```

Comment:
These locks can be removed if printing is done with a single function call (format can be the same, just pack everything into one call). Also, performance measurement mode would impact less other functionality if it would not use the same, global lock.